### PR TITLE
#149 color column

### PR DESCRIPTION
**Pull request by joshxg**
Partial #149 

- Adding just the color menu component and a circle svg (I used american spelling - thought `<ColourMenu color="x"/>` was a bit weird 
image
- although this features seems lower in priority, I thought doing this would give me some insight into doing mutations with the data table
- I think it did! 
- Need to pass a callback from `ListView`, down to the `Cell` level - i.e. `changeTransactionColor`.
- On mobile, we do this by passing a `getCallback` function to `DataTableRow`, which the `Row` calls and passes to the `Cell`, depending on the `CellKey`.
- With our current setup, we can pass into the `column` object passed into `useColumns` the callback for that column which does seem simpler

### CODE REVIEW

File: packages/common/src/ui/components/menus/ColorMenu/ColorMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const defaultColors = [
+  { hex: '#004fc4', name: 'blue' },
+  { hex: '#05a660', name: 'green' },
+  { hex: '#ff3b3b', name: 'red' },
+  { hex: '#fc0', name: 'yellow' },
+  { hex: '#00b7c4', name: 'aqua' },
+  { hex: '#8f90a6', name: 'grey' },
+];
```
Comment (joshxg) [inline packages/common/src/ui/components/menus/ColorMenu/ColorMenu.tsx:24]: Making this have a max width for when you have many options, with some line wrapping would have been good

File: packages/common/src/ui/icons/Circle.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import SvgIcon, { SvgIconProps } from '@material-ui/core/SvgIcon';
+
+export const Circle = (props: SvgIconProps): JSX.Element => {
+  return (
+    <SvgIcon
+      sx={{ overflow: 'visible' }}
```
Comment (joshxg) [inline packages/common/src/ui/icons/Circle.tsx:7]: I took the feather icons circle svg which keeps getting cut off by like 1px 😠

**Comment by joshxg** [comment]
Looking at this now, I should have made the `ColorMenu` component and probably a `ColorMenuCircleButton` component instead 🤔

File: packages/common/src/ui/icons/Icon.stories.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ import { Stock } from './Stock';
 import { Suppliers } from './Suppliers';
 import { Tools } from './Tools';
 import { Translate } from './Translate';
-
+import { Circle } from './Circle';
```
Comment (mark-prins) [inline packages/common/src/ui/icons/Icon.stories.tsx:29]: :nit would you mind putting this in alphabetical order?

**Review by mark-prins — APPROVED** [review]
that's awesome!

File: packages/common/src/ui/icons/Icon.stories.tsx (hunk 3 above)
Comment (joshxg) [reply packages/common/src/ui/icons/Icon.stories.tsx:29]: To be honest I tried using my alphabetical order vscode extension and it didn't work so I gave up, haha!!

File: packages/common/src/ui/icons/Circle.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import SvgIcon, { SvgIconProps } from '@material-ui/core/SvgIcon';
+
+export const Circle = (props: SvgIconProps): JSX.Element => {
+  return (
+    <SvgIcon sx={{ overflow: 'visible' }} viewBox="0 0 21 20" {...props}>
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/common/src/ui/icons/Circle.tsx]: you're a wizard 🧙

**Pull request merged**